### ISSUE TITLE
iptables backend wrongly symlinked

### ISSUE THREAD

**Issue by yhaenggi** [rajch/weave#2]
With IPTABLES_BACKEND you can choose between iptables/nftables backend. The current container symlinks /sbin/iptables* to nft

**Comment by rajch** [comment]
Great catch! How did you notice that?

To decide what to do about this, I have two observations:

* Since version 1.19, Alpine Linux defaults to the nft backend, and doesn't even install the legacy backend in the standard `iptables` package (See their [release](https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.19.0)). If we want to continue to provide the legacy backend as an option, we will have to add their `iptables-legacy` package as part of out build porcess

* The `launch.sh` script of weave-kube assumes that the default backend is legacy, and only attempts to change it if the `IPTABLES_BACKEND` variable is set to 'nft'.  If we want to continue to provide the legacy backend as an option, it will need to be modified to change the backend to legacy of `IPTABLES_BACKEND` variable is set to 'nft'.

So, decision time. Should we even try to continue support for the legacy backend, or should we just remove the choice and the option?

**Comment by yhaenggi** [comment]
It didnt work on one of my clusters where other things rely on iptables. I saw the iptable chains exist but with no rules.

For now, until iptables is gone, i'd say add it. That way it matches how it was before.

**Comment by rajch** [comment]
Okay. On it.

**Comment by rajch** [comment]
I've made an out-of-band release, with the proposed changes. To install it, obtain a manifest using:

```
curl -Lo net-2.8.4-beta1.yaml https://reweave.azurewebsites.net/k8s/v1.29/net.yaml?version=2.8.4-beta1
```

Then install it on kubernetes using:

```
kubectl apply -f  net-2.8.4-beta1.yaml
```
If the cluster already had an older weave add-on running, you will have to reboot the nodes.

By default, the launch script of the weave pods tries to auto-detect the available iptables back-end. If it detects nft,  it changes the `/sbin/iptables{,-save,-restore}` symlinks to use that. If it detects legacy (or the `IPTABLES_BACKEND` environment variable is set to anything but legacy), it does nothing. Using the 2.8.4-beta1 image, which has the `/sbin/iptables{,-save,-restore}` symlinks pointing to legacy by default, this will restore the old behaviour.

You can search the logs of the weave pods for the string "iptables backend mode:". This will show the mode that was autodetected or set by the `IPTABLES_BACKEND` environment variable.

Turns out none of the machines or VMs in by test lab have the legacy backend. I'll get some new ones, but meanwhile, if will be nice if you could take a look too.

**Comment by rajch** [comment]
As per my tests, this beta worked fine. @yhaenggi , I hope it worked for you too. I will be merging these changes, and releasing a new version soon.

**Comment by rajch** [comment]
Corrected in 664e361a . Published in v2.8.4, closing the issue now.
